subprojects/core/src/main/java/org/gradle/process/internal/worker/child/SystemApplicationClassLoaderWorker.java

```diff
@@ -110,13 +132,47 @@ private void configureLogging(LoggingManagerInternal loggingManager, ObjectConne
         loggingManager.addOutputEventListener(new WorkerLogEventListener(workerLoggingProtocol));
     }
 
-    MessagingServices createClient() {
-        return new MessagingServices();
+    private void configureWorkerProcessInfoEvents(WorkerServices services, ObjectConnection connection) {
+        connection.useParameterSerializers(WorkerProcessInfoSerializer.create());
+        final WorkerProcessInfoProtocol workerProcessInfoProtocol = connection.addOutgoing(WorkerProcessInfoProtocol.class);
+        services.get(MemoryManager.class).addListener(new JvmMemoryStatusListener() {
+            @Override
+            public void onJvmMemoryStatus(JvmMemoryStatus jvmMemoryStatus) {
+                workerProcessInfoProtocol.sendJvmMemoryStatus(jvmMemoryStatus);
+            }
+        });
     }
 
     LoggingManagerInternal createLoggingManager() {
         LoggingManagerInternal loggingManagerInternal = LoggingServiceRegistry.newEmbeddableLogging().newInstance(LoggingManagerInternal.class);
         loggingManagerInternal.captureSystemSources();
         return loggingManagerInternal;
     }
+
+    private static class WorkerServices extends DefaultServiceRegistry implements Stoppable {
+        public WorkerServices(ServiceRegistry... parents) {
+            super(parents);
+        }
+
+        ListenerManager createListenerManager() {
+            return new DefaultListenerManager();
+        }
+
+        OsMemoryInfo createOsMemoryInfo() {
+            return new DisabledOsMemoryInfo();
+        }
+
+        JvmMemoryInfo createJvmMemoryInfo() {
+            return new DefaultJvmMemoryInfo();
+        }
+
+        MemoryManager createMemoryManager(OsMemoryInfo osMemoryInfo, JvmMemoryInfo jvmMemoryInfo, ListenerManager listenerManager, ExecutorFactory executorFactory) {
+            return new DefaultMemoryManager(osMemoryInfo, jvmMemoryInfo, listenerManager, executorFactory);
+        }
+
+        @Override
+        public void stop() {
```

Comment:
Doesn't look like this is ever used. Also, I don't think it's necessary as the registry is `Closeable` so the thing that might stop this can just close it instead.